sdks/python/apache_beam/ml/rag/enrichment/bigquery_vector_search.py

```diff
@@ -140,48 +184,66 @@ def format_query(self, chunks: List[Chunk]) -> str:
             FROM UNNEST(metadata) 
             WHERE key = search_key AND value = search_value
         ));
+    """ if self.metadata_restriction_template else ""
+
+    # Group chunks by their metadata conditions
+    condition_groups = {}
+    if self.metadata_restriction_template:
+      for chunk in chunks:
+        condition = self.metadata_restriction_template.format(**chunk.metadata)
+        if condition not in condition_groups:
+          condition_groups[condition] = []
+        condition_groups[condition].append(chunk)
```

Comment:
Nit: this can be cleaned up by using defaultdict - https://www.geeksforgeeks.org/defaultdict-in-python/